plugins/reportal/src/azkaban/reportal/util/ReportalUtil.java

```diff
@@ -57,4 +62,63 @@ public static IStreamProvider getStreamProvider(String fileSystem) {
 
     return sortedNodes;
   }
+
+  /**
+   * Get runtime variables to be set in unscheduled mode of execution.
+   * Returns empty list, if no runtime variable is found
+   *
+   * @param variables
+   * @return
+   */
+  public static List<Variable> getRunTimeVariables(
+    Collection<Variable> variables) {
+    List<Variable> runtimeVariables =
+      ReportalUtil.getVariablesByRegex(variables, "(?!(^"
+        + Reportal.REPORTAL_CONFIG_PREFIX + ")).+");
+
+    return runtimeVariables;
+  }
+
+  /**
+   * Shortlist variables which match a given regex. Returns empty props, if no
+   * eligible variable is found
+   *
+   * @param variables
+   * @param regex
+   * @return
+   */
+  public static List<Variable> getVariablesByRegex(
+    Collection<Variable> variables, String regex) {
+    List<Variable> shortlistedVariables = new ArrayList<Variable>();
+    if (variables != null || regex == null) {
+      for (Variable var : variables) {
+        if (var.getTitle().matches(regex)) {
+          shortlistedVariables.add(var);
+        }
+      }
+    }
+    return shortlistedVariables;
+  }
+
+  /**
+   * Shortlist variables which match a given prefix. Returns empty props, if no
+   * eligible variable is found.
+   *
+   * @param variables
+   *          variables to be processed
+   * @param prefix
+   *          prefix to be matched
+   * @return a map with shortlisted variables and prefix removed
+   */
+  public static Map<String, String> getVariableMapByPrefix(
+    Collection<Variable> variables, String prefix) {
+    Map<String, String> shortlistMap = new HashMap<String, String>();
+    if (prefix != null) {
```

Comment:
should you be checking for 
variable == null
?

for clarity's sake even if it might be logically correct and checked in getVariablesByRegex()